plugins/woocommerce/src/StoreApi/Routes/V1/Checkout.php

```
@@ -355,6 +376,16 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 			$this->process_without_payment( $request, $payment_result );
 		}
 
+		wc_log_order_step(
+			'[Store API #10] Order processed',
+			array(
+				'order_id'               => $this->order->get_id(),
+				'processed_with_payment' => $this->order->needs_payment() ? 'yes' : 'no',
+				'payment_status'         => $payment_result->status,
+			),
+			true
+		);
+
```

I suggest you just go inside process_without_payment and process_payment and do the same as shortcode.

---

I'll open the PR up for review, tag some more eyes, and then will look into these 👍🏼

---

Agree with Nadir. See my previous comment about going deeper.

---

Just went inside those. They seem pretty robust yet with thrown errors, apart from that, it's only setters, not much can go wrong there. I think there won't be a silent error there, if we see logs dropping off just before it we may go deeper in a second iteration.

---

At least for Store API. I also added backtrace and a last error entry.